astroid/rebuilder.py

```diff
@@ -357,7 +357,11 @@ def visit_comprehension(self, node, parent):
         newnode.postinit(self.visit(node.target, newnode),
                          self.visit(node.iter, newnode),
                          [self.visit(child, newnode)
-                          for child in node.ifs])
+                          for child in node.ifs],
+                         nodes.Const(getattr(node, 'is_async', None),
```

Comment:
Check my previous comment regarding ``is_async``. Sorry, I think I misled you in a different PR, with a similar use case.